scripts/manage-db.py

```diff
@@ -86,13 +87,53 @@ def cleanup_releases_history(trans, dryrun=True):
     print "Total Deleted: %d" % total_deleted
 
 
-def extract_active_data(url, dump_location='dump.sql'):
+def _extract_partials_generator(json_object):
+    """
+    Returns a generator that contains all value objects having key as 'partials'
+    """
+    if isinstance(json_object, dict):
+        for key, value in json_object.iteritems():
+            if "partial" == key or "partials" == key:
+                yield value
+            else:
+                for child_val in _extract_partials_generator(value):
+                    yield child_val
+
+    elif isinstance(json_object, list):
+        for item in json_object:
+            for value in _extract_partials_generator(item):
+                yield value
+    else:
+        pass
+
+
+def _extract_partial_release_names(release_data):
+    """
+    Extracts all partial release's names from the release_data.
+    :param release_data: input release data for processing
+    :return partial_release_names : a set of extracted partial release names
+    """
+    partial_release_names = set()
+    for item in _extract_partials_generator(release_data):
+        if isinstance(item, dict):
+            map(partial_release_names.add, list(item.keys()))
+        elif isinstance(item, list):
+            for list_item in item:
+                if 'from' in list_item:
+                    partial_release_names.add(str(list_item['from']))
+        else:
+            pass
+    return partial_release_names
```

Comment:
For posterity, we talked about this not working for certain types of releases on IRC: https://gist.github.com/mozbhearsum/2f9669007476a960fa4a51d50bc6e0c4